components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/EmailFormatValidator.java

```diff
@@ -0,0 +1,115 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.input.validation.mgt.model.validators;
+
+import org.wso2.carbon.identity.input.validation.mgt.exceptions.InputValidationMgtClientException;
+import org.wso2.carbon.identity.input.validation.mgt.model.Property;
+import org.wso2.carbon.identity.input.validation.mgt.model.ValidationContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.DEFAULT_EMAIL_REGEX_PATTERN;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.ENABLE;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.USERNAME;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.ErrorMessages.ERROR_VALIDATION_EMAIL_FORMAT_MISMATCH;
+
+/**
+ * Email format validator.
+ */
+public class EmailFormatValidator extends AbstractRulesValidator {
+
+    private final List<String> allowedFields = new ArrayList<String>() {{
+        add(USERNAME);
+    }};
+
+    @Override
+    public boolean isAllowedField(String field) {
+
+        return allowedFields.contains(field);
+    }
+
+    /**
+     * Validate the string against the validation criteria.
+     *
+     * @param context   Validation Context.
+     * @return boolean
+     * @throws InputValidationMgtClientException Error when string does not satisfy the validation criteria
+     */
+    @Override
+    public boolean validate(ValidationContext context) throws InputValidationMgtClientException {
+
+        String value = context.getValue();
+        String field = context.getField();
+        Map<String, String> attributesMap = context.getProperties();
+        String emailRegEx = DEFAULT_EMAIL_REGEX_PATTERN;
+
+        // Check whether value satisfies the email format criteria.
+        if (attributesMap.containsKey(ENABLE)) {
+            if (Boolean.parseBoolean(attributesMap.get(ENABLE)) && value != null && !value.matches(emailRegEx)) {
+                throw new InputValidationMgtClientException(ERROR_VALIDATION_EMAIL_FORMAT_MISMATCH.getCode(),
+                        ERROR_VALIDATION_EMAIL_FORMAT_MISMATCH.getMessage(),
+                        String.format(ERROR_VALIDATION_EMAIL_FORMAT_MISMATCH.getDescription(), field, emailRegEx));
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Get list of supported properties for the validator.
+     *
+     * @return  List<Property>
+     */
+    @Override
+    public List<Property> getConfigurationProperties() {
+
+        List<Property> configProperties = new ArrayList<>();
+        int parameterCount = 0;
+
+        Property enable = new Property();
```

Comment:
Modify the name

Reply:
Addressed with commit 6e4ccba6376101562c801f735b660c94f4a280b2